metrics_batcher.go

```diff
@@ -31,19 +34,24 @@ type metricsBatcher struct {
 	// Counts all dropped TimeSeries by this metricsBatcher.
 	droppedTimeSeries int
 
-	workers   []*worker
-	curWorker int
-	// respsChan is shared between metricsBatcher and worker goroutines.
+	workers []*worker
+	// reqsChan, respsChan and wg are shared between metricsBatcher and worker goroutines.
+	reqsChan  chan *monitoringpb.CreateTimeSeriesRequest
 	respsChan chan *response
-
-	mc *monitoring.MetricClient
+	wg        *sync.WaitGroup
 }
 
 func newMetricsBatcher(ctx context.Context, projectID string, numWorkers int, mc *monitoring.MetricClient) *metricsBatcher {
+	if numWorkers < minNumWorkers {
+		numWorkers = minNumWorkers
+	}
 	workers := make([]*worker, 0, numWorkers)
+	reqsChan := make(chan *monitoringpb.CreateTimeSeriesRequest, numWorkers)
```

Comment:
Maybe start with `3 * numWorkers`. This will allow us to prepare more batches for the worker threads :)

Reply:
SG, done.

Reply:
Sorry, probably is better to have:

max(numWorkers, 5) -> so to have at least 5 pending batches :)

Reply:
Oh I missed this one - will fix in another PR.